doc/data_structures.rst

```diff
@@ -277,50 +284,58 @@ metadata is shown below:
 When adding a scalar or string metadata value, the value
 is copied by the ``DataSet``, and as a result, the user
 does not need to ensure that the metadata values provided
-are still in memory after they have been added.
+are still in-memory. In other words, 
+the ``DataSet`` handles the memory management of these metadata values, 
+and you don't need to retain or manage the original copies separately 
+once they have been included in the ``DataSet`` object.
 Additionally, multiple metadata values can be added to a
-single field, and the default behavior is to append the value to the
-existing field.  In this way, the ``DataSet`` metadata supports
-one-dimensional arrays, but the entries in the array must be added
-iteratively by the user.  Also, note that in the above C++ example,
+single field name, and the default behavior is to append the value to the
+field name if it exists, create if not. This behavior allows the ``DataSet`` metadata 
+to function like one-dimensional arrays. However, if you would like to add
+multiple metadata values to a field name, you will need to add them one by 
+one in an iterative manner.
```

Comment:
Delete "However, if you would like to add
multiple metadata values to a field name, you will need to add them one by 
one in an iterative manner."